snowflake/cmd/trafficDrop.go

```diff
@@ -0,0 +1,294 @@
+	Short: "Run the abnormal traffic drop detection UDF in Snowflake",
+	Long: `This command runs the abnormal traffic drop detection UDF in
+Snowflake. You need to bring your own Snowflake account and run the onboard
+command first.
+
+Run abnormal traffic drop detection with default configuration on database ANTREA_C9JR8KUKUIV4R72S:
+"theia-sf traffic-drop --database-name ANTREA_C9JR8KUKUIV4R72S"
+
+The "traffic-drop" command requires a Snowflake warehouse to run abnormal
+traffic drop detection UDF in Snowflake. By default, it will create a temporary
+one.
+You can also bring your own by using the "--warehouse-name" parameter.
+`,
+	Args: cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		jobType, _ := cmd.Flags().GetString("type")
+		if jobType != "initial" && jobType != "periodical" {
+			return fmt.Errorf("invalid --type argument")
+		}
+
+		start, _ := cmd.Flags().GetString("start")
+		end, _ := cmd.Flags().GetString("end")
+		startTs, _ := cmd.Flags().GetString("start-ts")
+		endTs, _ := cmd.Flags().GetString("end-ts")
+		clusterUUID, _ := cmd.Flags().GetString("cluster-uuid")
+		databaseName, _ := cmd.Flags().GetString("database-name")
+		warehouseName, _ := cmd.Flags().GetString("warehouse-name")
+		functionVersion, _ := cmd.Flags().GetString("udf-version")
+		waitTimeout, _ := cmd.Flags().GetString("wait-timeout")
+		waitDuration, err := time.ParseDuration(waitTimeout)
+		if err != nil {
+			return fmt.Errorf("invalid --wait-timeout argument, err when parsing it as a duration: %v", err)
+		}
+
+		query, err := buildTrafficDropUdfQuery(jobType, start, end, startTs, endTs, clusterUUID, databaseName, functionVersion)
+		if err != nil {
+			return err
+		}
+		ctx, cancel := context.WithTimeout(context.Background(), waitDuration)
+		defer cancel()
+		rows, err := udfs.RunUdf(ctx, logger, query, databaseName, warehouseName)
+		if err != nil {
+			return fmt.Errorf("error when running traffic drop UDF: %w", err)
+		}
+		defer rows.Close()
+
+		var detectionID string
+		var timeCreated string
+		var endpoint string
+		var direction string
+		var avgDrop float32
+		var stdevDrop float32
+		var anomalyDropDate string
+		var anomalyDropNumber float32
+		for rows.Next() {
+			if err := rows.Scan(
+				&jobType,
+				&detectionID,
+				&timeCreated,
+				&endpoint,
+				&direction,
+				&avgDrop,
+				&stdevDrop,
+				&anomalyDropDate,
+				&anomalyDropNumber,
+			); err != nil {
+				return fmt.Errorf("invalid row: %w", err)
+			}
+			fmt.Printf("endpoint: %s, direction: %s, avgDrop: %f, stdevDrop: %f, anomalyDropDate: %s, anomalyDropNumber: %f\n",
+				endpoint,
+				direction,
+				avgDrop,
+				stdevDrop,
+				anomalyDropDate[:len(anomalyDropDate)-10],
+				anomalyDropNumber)
+		}
+
+		return nil
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(trafficDropCmd)
+
+	trafficDropCmd.Flags().String("type", "initial", "Type of abnormal traffic drop detection job (initial|periodical), we only support initial jobType for now")
```

Comment:
we never enforce that jobType is initial anywhere. Should we just remove that parameter altogether?

Reply:
Thanks pointing out! I feel jobtype still useful in the future, so I add an assert in UDF to enforce it.